### DISCUSSION TITLE
SMPClient set timout to avoid java.net.SocketTimeoutException

### DISCUSSION THREAD

**Discussion by iansmirlis** [phax/phase4#242] [General]
Hello

Sorry if this is already answered, I couldn't find something related

We have a receiver with some slow SMP server, and we often get timeout on http query

Is there a way to increase timeout and/or retries on SMPClientReadOnly object?

Thanks!

**Comment by phax** [comment]
No worries,
The easiest way to extend the timeout settings is via the `HttpClientSettings` of the SMP Client you are using:
Either via ` aSMPClient.withHttpClientSettings (hcs -> ...);` or via `aSMPClient.httpClientSettings ().set (...);`

**Comment by iansmirlis** [comment]
Thanks for your prompt reply.

I noticed that in some cases I also get connection refused errors from their SMP. I guess there is no way to give a retry parameter here

**Reply by phax** [reply]
Well, it has some retry methods as well: `setRetryCount`, `setRetryInterval` and `setRetryAlways`

**Reply by iansmirlis** [reply]
Great thanks!
